From 0b320482c71b10a2ec9b2a99f98c7487e72232ce Mon Sep 17 00:00:00 2001
From: Hugo Benichi <hugobenichi@google.com>
Date: Fri, 7 Sep 2018 12:27:03 +0900
Subject: [PATCH] CHROMIUM: Revert "xfrm: Refuse to insert 32 bit userspace
 socket policies on 64 bit systems"

For the same reason commit 74005991b78a ("xfrm: Do not parse 32bits
compiled xfrm netlink msg on 64bits host") is reverted in our local
source tree, we also revert commit dffe655ddb48 ("xfrm: Refuse to
insert 32 bit userspace socket policies on 64 bit systems") to work
around user/kernel ABI incompatibility in xfrm.

BUG=b:114323255
BUG=b:112641692
TEST=- Compiled, flashed grunt, booted.
     - Verified that netd's XfrmController starts correctly (logcat).
     - Verified that android.net.cts.IpSecBaseTest and
     android.net.cts.IpSecManagerTest pass.
Change-Id: I27cd1fe7f634fc034dd3a6cc29a72cd34de4c33b
Signed-off-by: Hugo Benichi <hugobenichi@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/1157958
Commit-Ready: Hugo Benichi <hugobenichi@google.com>
Tested-by: Hugo Benichi <hugobenichi@google.com>
Reviewed-by: Dmitry Torokhov <dtor@chromium.org>

[rebase52(rrangel): Context conflict. #ifdef removed upstream]
Signed-off-by: Raul E Rangel <rrangel@chromium.org>
---
 net/xfrm/xfrm_state.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/net/xfrm/xfrm_state.c b/net/xfrm/xfrm_state.c
index c6f3c4a1bd99..c023b27c70e1 100644
--- a/net/xfrm/xfrm_state.c
+++ b/net/xfrm/xfrm_state.c
@@ -2266,9 +2266,6 @@ int xfrm_user_policy(struct sock *sk, int optname, u8 __user *optval, int optlen
 	struct xfrm_mgr *km;
 	struct xfrm_policy *pol = NULL;
 
-	if (in_compat_syscall())
-		return -EOPNOTSUPP;
-
 	if (!optval && !optlen) {
 		xfrm_sk_policy_insert(sk, XFRM_POLICY_IN, NULL);
 		xfrm_sk_policy_insert(sk, XFRM_POLICY_OUT, NULL);
-- 
2.23.0.rc1.153.gdeed80330f-goog

